### DISCUSSION TITLE
NovelRT::Ecs::TransformComponent: scale inheritance in the scene hierarchy is arithmetic instead of geometric

### DISCUSSION THREAD

**Discussion by dndn1011** [novelrt/NovelRT#595] [Q&A]
Hi,

I was taking a look at the engine and noticed that scales are combined by addition, instead of multiplication.

`            positionAndLayer += other.positionAndLayer;
            rotationInRadians += other.rotationInRadians;
            scale += other.scale;`

Usually, scale is a multiplier, is this by intention?  

-Dino

**Comment by capnkenny** [comment]
Hi Dino!

I'll let @RubyNova confirm or correct me since it is his domain area, however I do believe this is intentional.
`TransformComponent.scale` is treated as a multiplier insofar as it is multiplied to the generated `scaleValue` for scaling the matrix of the object being manipulated, and can be added to / removed from to adjust the scaling of said object. For more detail, you can look [here](https://github.com/novelrt/NovelRT/blob/97c1823217338e18b850c79f4719931699291817/src/NovelRT/Ecs/Graphics/DefaultRenderingSystem.cpp#L341) as a reference to how the current `DefaultRenderingSystem` currently handles it.

-Kenny
